linux/ibus-keyman/meson.build

```diff
@@ -8,9 +8,17 @@ conf = configuration_data()
 ibus =           dependency('ibus-1.0',             version: '>= 1.2.0')
 gtk =            dependency('gtk+-3.0',             version: '>= 2.4')
 json_glib =      dependency('json-glib-1.0',        version: '>= 1.0')
-systemd  =       dependency('libsystemd')
 icu =            dependency('icu-i18n')
 
+systemd  =       dependency('libsystemd', 'libelogind', required: false)
+if systemd.found()
+  conf.set('DBUS_IMPLEMENTATION', 'SYSTEMD')
+else
+  # Gentoo can use libelogind or basu
+  systemd = dependency('basu')
+  conf.set('DBUS_IMPLEMENTATION', 'BASU')
+endif
```

Comment:
Is there a failure if none of these are available?

Reply:
Yes, the `dependency` function has an implicit `required: true` parameter. But probably good to add it anyways to make it clearer.